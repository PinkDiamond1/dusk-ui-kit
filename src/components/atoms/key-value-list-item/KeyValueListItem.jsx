/* @flow */
import * as React from 'react'

type Props = {
  keyTitleClass?: string,
  keyValueClass?: string,
  keyTitle: string,
  keyValue: string,
}

const KeyValueListItem = (props: Props): React.Element<*> => {
  const { keyTitle, keyValue, keyTitleClass, keyValueClass } = props
  return (
    <>
      <dt
        className={keyTitleClass}
        style={{
          fontWeight: 400,
          paddingLeft: 0,
        }}
      >
        {keyTitle}
      </dt>
      <dd
        className={keyValueClass}
        style={{
          textAlign: 'right',
          fontWeight: 500,
          paddingRight: 0,
        }}
      >
        {keyValue}
      </dd>
    </>
  )
}

KeyValueListItem.defaultProps = {
  keyTitleClass: 'col-6',
  keyValueClass: 'col-6',
  keyTitle: '',
  keyValue: '',
}

export default KeyValueListItem
