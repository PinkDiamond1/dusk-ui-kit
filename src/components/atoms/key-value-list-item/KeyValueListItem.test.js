import React from 'react'
import renderer from 'react-test-renderer'

import KeyValueListItem from './'

it('test to see if the KeyValueListItem renders correctly', () => {
  const tree = renderer.create(<KeyValueListItem>Smoke Test</KeyValueListItem>).toJSON()
  expect(tree).toMatchSnapshot()
})
