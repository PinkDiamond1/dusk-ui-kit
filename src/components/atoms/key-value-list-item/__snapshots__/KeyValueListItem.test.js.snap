// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the KeyValueListItem renders correctly 1`] = `
Array [
  <dt
    className="col-6"
    style={
      Object {
        "fontWeight": 400,
        "paddingLeft": 0,
      }
    }
  >
    
  </dt>,
  <dd
    className="col-6"
    style={
      Object {
        "fontWeight": 500,
        "paddingRight": 0,
        "textAlign": "right",
      }
    }
  >
    
  </dd>,
]
`;
